sqlmesh/dbt/adapter.py

```diff
@@ -355,3 +362,54 @@ def _normalize(self, input_table: exp.Table) -> exp.Table:
             normalized_table.set("db", normalized_table.this)
             normalized_table.set("this", None)
         return normalized_table
+
+
+class CachingAdapter:
+    """Proxy that enhances engine adapter methods with dbt cache updating logic."""
+
+    def __init__(
+        self,
+        engine_adapter: EngineAdapter,
+        cache: t.Dict[exp.Table, t.List[BaseRelation]],
+        relation_type: t.Type[BaseRelation],
+    ):
+        self.engine_adapter = engine_adapter
+        self.cache = cache
+        self.relation_type = relation_type
+
+    def __getattr__(self, attr: str) -> None:
+        return getattr(self.engine_adapter, attr)
+
+    def drop_table(self, *args: t.Any, **kwargs: t.Any) -> None:
+        self.engine_adapter.drop_table(*args, **kwargs)
+        self.cache.clear()
+
+    def drop_schema(self, *args: t.Any, **kwargs: t.Any) -> None:
+        self.engine_adapter.drop_schema(*args, **kwargs)
+        self.cache.clear()
+
+    def drop_view(self, *args: t.Any, **kwargs: t.Any) -> None:
+        self.engine_adapter.drop_view(*args, **kwargs)
+        self.cache.clear()
+
+    def create_table(self, table_name: TableName, *args: t.Any, **kwargs: t.Any) -> None:
+        self.engine_adapter.create_table(table_name, *args, **kwargs)
+        self._cache_relation(table_name)
+
+    def create_view(self, view_name: TableName, *args: t.Any, **kwargs: t.Any) -> None:
+        self.engine_adapter.create_view(view_name, *args, **kwargs)
+        self._cache_relation(view_name)
+
+    def _cache_relation(self, relation_name: TableName) -> None:
+        table = exp.to_table(relation_name)
+        schema = to_schema(table)
+        relation = self.relation_type.create(
+            database=table.catalog or None, schema=table.db, identifier=table.name
```

Comment:
Should it be `default_catalog` instead of `None`?

Reply:
Not sure - I was trying to follow what is done in https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/dbt/adapter.py#L334 but we can double-check this